hack/generated/pkg/genruntime/convertible_status.go

```diff
@@ -0,0 +1,14 @@
+/*
+ * Copyright (c) Microsoft Corporation.
+ * Licensed under the MIT license.
+ */
+
+package genruntime
+
+type ConvertibleStatus interface {
```

Comment:
I was just suggesting that since the size of possible Specs is large and the size of possible "correct types" to pass to one of these functions is small/one, that we're always going to run a large risk of runtime error regardless and so we could optimize for slightly less code if we wanted to.